src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,341 @@
+            if (command.equals("bye")) {
+                printGoodbyeMessage();
+                isFinished = true;
+            } else if (command.equals("list")) {
+                printListOfTask(tasks);
+            } else {
+                updateListOfTask(command, tasks);
+                try {
+                    saveListToFile(tasks, pathOfDataFile);
+                } catch (IOException e) {
+                    System.out.println("I/O error has occurred");
+                }
+            }
+        }
+    }
+
+    private static String loadListFromFile(ArrayList<Task> tasks) {
+        checkIfDirectoryExists();
+        String pathOfDataFile = "data/duke.txt";
+        try {
+            File dataFile = checkIfFileExists(pathOfDataFile);
+            readListFromFile(tasks, dataFile);
+        } catch (IOException e) {
+            System.out.println("I/O error has occurred");
+        }
+
+        if (tasks.size() > 0) {
+            printListOfTask(tasks);
+        }
+        return pathOfDataFile;
+    }
+
+    private static void readListFromFile(ArrayList<Task> tasks, File dataFile) throws FileNotFoundException {
+        Scanner fileReader = new Scanner(dataFile);
+        if (fileReader.hasNextLine()) {
+            fileReader.nextLine();
+        }
+        while (fileReader.hasNextLine()) {
+            String[] sectionsOfCurrentLine;
+            String currentLine = fileReader.nextLine();
+            sectionsOfCurrentLine = currentLine.split("\\|");
+            String typeOfSavedTask = sectionsOfCurrentLine[0].trim();
+            String statusOfSavedTask = sectionsOfCurrentLine[1].trim();
+            String informationOfSavedTask = sectionsOfCurrentLine[2].stripLeading();
+            Task newTask;
+            switch (typeOfSavedTask) {
+            case "T":
+                newTask = new ToDo(informationOfSavedTask);
+                addTaskToList(tasks, statusOfSavedTask, newTask);
+                break;
+            case "D":
+                String detailsOfSavedDeadline = sectionsOfCurrentLine[3].trim();
+                newTask = new Deadline(informationOfSavedTask, detailsOfSavedDeadline);
+                addTaskToList(tasks, statusOfSavedTask, newTask);
+                break;
+            case "E":
+                String detailsOfSavedEvent = sectionsOfCurrentLine[3].trim();
+                newTask = new Event(informationOfSavedTask, detailsOfSavedEvent);
+                addTaskToList(tasks, statusOfSavedTask, newTask);
+                break;
+            default:
+                System.out.println("Unknown task type");
+                break;
+            }
+        }
+    }
+
+    private static void addTaskToList(ArrayList<Task> tasks, String statusOfSavedTask, Task newTask) {
+        if (statusOfSavedTask.equals("1")) {
+            newTask.markAsDone();
+        }
+        tasks.add(newTask);
+    }
+
+    private static File checkIfFileExists(String pathOfDataFile) throws IOException {
```

Comment:
The name of this method may not be very clear as I would expect it only to check if the file exists but actually it creates a new file.